cartridges/app_storefront_address_extend/cartridge/forms/default/address.xml

```diff
@@ -0,0 +1,62 @@
+<?xml version="1.0"?>
+<form xmlns="http://www.demandware.com/xml/form/2008-04-19">
+
+    <field formid="companyName" label="label.input.companyname" type="string" mandatory="false"
+        missing-error="error.message.required" />
+    <field formid="vat" label="label.input.vat" type="string" mandatory="false"
```

Comment:
You can add a regex for the VAT number. Starting with 2 alpha symbols, followed by up to 10 digits.